app/src/main/java/space/taran/arknavigator/mvp/model/repo/PlainResourcesIndex.kt

```diff
@@ -72,13 +81,19 @@ class PlainResourcesIndex internal constructor (
     }
 
     internal suspend fun reindexRoot(diff: Difference) = withContext(Dispatchers.IO) {
+        val savedDeletedMetas = mutableMapOf<Path, ResourceMeta?>()
         diff.deleted.forEach {
             pathById[metaByPath[it]!!.id]
+            savedDeletedMetas[it] = metaByPath[it]
```

Comment:
Just delete preview here. No need to collect items and then do it in one go since all data resides in memory here.